python/ray/util/sgd/v2/trainer.py

```diff
@@ -584,3 +562,74 @@ def default_resource_request(cls,
             return PlacementGroupFactory(bundles, strategy="PACK")
 
     return SgdTrainable
+
+
+class ExecutableTrainer:
+    """A class for stateful execution on an executable class.
+
+    This allows you to pass in a class definition to be instantiated on each
+    worker, and then execute methods on this class on each worker.
+
+    This should not be instantiated directly and instead should be used only
+    from the output of ``trainer.run_executable``.
+    """
+
```

Comment:
I'm still not quite sure what this does. How do you use it? What is the behavior? How do you shut it down?

Reply:
To answer your question, it's a way to run methods of the passed in executable class on multiple workers. You would use it by first calling the appropriate execute method (execute, execute_single, etc.) and then call the executable class's method on that. (like in the PR description). This class doesn't need to be shutdown. Once the `Trainer` is shutdown , then any subsequent operations on this will result in an InactiveWorkerGroup error.

Reply:
I'll add this to the docstring.